docs/cmake-presets.md

```diff
@@ -0,0 +1,272 @@
+
+**Select your compilers**
+C and C++ compilers can be set with `cacheVariables.CMAKE_C_COMPILER` and `cacheVariables.CMAKE_CXX_COMPILER` in a Configure Preset. This is equivalent to passing `-D CMAKE_C_COMPILER=<value> and -D CMAKE_CXX_COMPILER=<value>` to CMake from the command line. For more information, see [CMAKE_<LANG>_COMPILER](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER.html#cmake-lang-compiler).
+
+You can specify the name of a compiler on your `PATH` or an environment variable that evaluates to the full path of a compiler. Full paths are discouraged so that the file will remain shareable.
+
+When you build with the Visual C++ toolset, CMake Tools automatically sources the environment from the latest version of the Visual Studio Build Tools installed on your system. You can specify a specific compiler version with the `toolset` option in `CMakePresets.json`. For more information, see [Configure Presets and Toolset Selection](https://cmake.org/cmake/help/latest/manual/cmake-toolchains.7.html).
+
+A preset that builds for 64-bit Windows with cl.exe and a Visual Studio generator might look like this.
+
+```json
+"architecture": {
+   "value": "x64",
+   "strategy": "set"
+},
+"cacheVariables": {
+   "CMAKE_BUILD_TYPE": "Debug",
+   "CMAKE_C_COMPILER": "cl",
+   "CMAKE_CXX_COMPILER": "cl",
+   "CMAKE_INSTALL_PREFIX": "${sourceDir}/out/install/${presetName}"
+ },
+ ```
+
+A preset that builds with GCC version 8 on Linux or macOS might look like this.
+
+```json
+"cacheVariables": {
+    "CMAKE_BUILD_TYPE": "Debug",
+    "CMAKE_INSTALL_PREFIX": "${sourceDir}/out/install/${presetName}",
+    "CMAKE_C_COMPILER": "gcc-8",
+    "CMAKE_CXX_COMPILER": "g++-8"
+}
+```
+
+You can also set compilers with a CMake toolchain file. Toolchain files can be set with `cacheVariables.CMAKE_TOOLCHAIN_FILE`, which is equivalent to passing `-D CMAKE_TOOLCHAIN_FILE=<value>` to CMake from the command line. A CMake toolchain file is most often used for cross-compilation. See [CMake toolchains](https://cmake.org/cmake/help/latest/manual/cmake-toolchains.7.html) for more information on authoring CMake toolchain files.
+
+## Select your generator
+
+Configure Preset templates default to the Visual Studio generator on Windows, and Ninja on Linux and macOS. You can specify a new generator with the generator option in a Configure Preset. This is equivalent to passing `-G` to CMake from the command line. See [CMake generators](https://cmake.org/cmake/help/latest/manual/cmake-generators.7.html#:~:text=A%20CMake%20Generator%20is%20responsible%20for%20writing%20the,what%20native%20build%20system%20is%20to%20be%20used) for more information.
+
+> [!Note] Windows developers: If Ninja is not installed with Visual Studio, then it will need to be on your `PATH`. Make sure to set `architecture.strategy` and `toolset.strategy` to external when building with a command-line generator like Ninja on Windows.
```

Comment:
...on your `PATH` or in the default installation folder.

Reply:
done